KEYCLOAK-8224 Client not found error message

slaskawi opened the pull request:
https://issues.jboss.org/browse/KEYCLOAK-8224

slaskawi committed:
KEYCLOAK-8224 Client not found error message

mposolda commented on services/src/main/java/org/keycloak/services/resources/admin/UserResource.java:
```diff
@@ -686,9 +686,13 @@ public Response executeActionsEmail(@QueryParam(OIDCLoginProtocol.REDIRECT_URI_P
-        if (client == null || !client.isEnabled()) {
+        if (client == null) {
+            throw new WebApplicationException(
+                ErrorResponse.error(clientId + " doesn't exist", Status.BAD_REQUEST));
```
Is it possible to change it to generic error message like "Client doesn't exists" and eventually add a warning/error message to the log with clientId? Just thinking about some potential attacks when clientId contains for example XSS script and the error will be then somehow displayed in the browser - you remember from security training :) AFAIK we changed most of error messages already to be "Generic" and not return the input, but not yet this one.

slaskawi replied:
Good one! I'm on it.

mposolda commented on services/src/main/java/org/keycloak/services/resources/admin/UserResource.java:
```diff
@@ -686,9 +686,13 @@ public Response executeActionsEmail(@QueryParam(OIDCLoginProtocol.REDIRECT_URI_P
+        if (!client.isEnabled()) {
             throw new WebApplicationException(
-                ErrorResponse.error(clientId + " not enabled", Status.BAD_REQUEST));
+                    ErrorResponse.error(clientId + " not enabled", Status.BAD_REQUEST));
```
Same as above

mposolda reviewed (commented):
@slaskawi Thanks for the PR! It it good, just added two minor comments about the error message.

One more thing, it seems that UserTest is failing in travis and looks like it is a regression of this?

slaskawi commented:
@mposolda Thanks for the review. I addressed both your comments and removed the test I added. It turns out that both paths (client not found and client disabled) are tested in `UserTest`.

mposolda commented on services/src/main/java/org/keycloak/services/resources/admin/UserResource.java:
```diff
@@ -686,9 +686,15 @@ public Response executeActionsEmail(@QueryParam(OIDCLoginProtocol.REDIRECT_URI_P
+                ErrorResponse.error("Client doesn't exist", Status.BAD_REQUEST));
+        }
+        if (!client.isEnabled()) {
+            logger.debugf("Client %s is not enabled exist", clientId);
```
@slaskawi Thanks! I've just noticed that there seems to be a small typo in the debug message here? Sorry to bother, I hope this one won't take you too much time to fix :)

slaskawi replied:
Fixed. Thanks for spotting this!

mposolda reviewed (approved):
@slaskawi Thanks!
